app/forms/form.rb

```diff
@@ -0,0 +1,37 @@
+class Form
+  include ActiveModel::Model
+  include ActiveModel::Validations
+
+  NotImplementedError = Class.new(StandardError)
+
+  attr_accessor :session
+  attr_reader :params, :question_name, :flow_name
+
+  def initialize(params, session, flow_name, question_name)
```

Comment:
I think the way `initialize` is used in the subclasses (reliance on `super`) could be improved. If the following is added to this class:
```ruby
    class << self
      attr_reader :flow_name, :node_name

      def part_of_flow(flow_name)
        @flow_name = flow_name
      end

      def form_for_node(node_name)
        @node_name = node_name
      end
    end

    delegate :flow_name, :node_name, to: :class
```

The flow and node could be set in each subclass like this and `initialize` would not have to be overwritten:
```ruby
class FooForm < Form
  part_of_flow :foo 
  form_for_node :bar 
 
   .....
```
With that in place you get:
```
  FooForm.flow_name => :foo 
  FooForm.node_name => :bar 
  foo = FooForm.new({}, {})
  foo.flow_name => :foo 
  foo.node_name => :bar 
```

Reply:
And to clarify, `initialize` within `Flow` would then be:
```ruby
  attr_accessor :session
  attr_reader :params

  def initialize(params, session)
    @params = params
    @session = session
  end
```

Reply:
Thanks @reggieb - Have updated as per your suggestion, but I didn't like your attr names (sorry!) - as this is all about `answers`.  I've also read [Yehua Katz on `<< self`](https://yehudakatz.com/2009/11/15/metaprogramming-in-ruby-its-all-about-the-self/) and think I need to investigate further!